hyperspy/tests/signal/test_eds_tem.py

```diff
@@ -265,6 +265,17 @@ def test_intensity(self):
             [0.5, 0.5],
             atol=1e-1)
 
+    def test_intensity_dtype_uint(self):
+        s = self.signal
+        s.data *= 1E5
+        s.change_dtype("uint")
+
+        np.testing.assert_allclose(
+            [i.data[0] for i in s.get_lines_intensity(
```

Comment:
You have to supply `background_windows` to trigger the issue that you attempt to fix.

Namely this would work:
```python
bw = s.estimate_background_windows()
np.testing.assert_allclose(
    [i.data[0] for i in s.get_lines_intensity(background_windows=bw)],
# other arguments / values
    )
```